databases/week3/databases_week3_mealsharing_DML.sql

```diff
@@ -0,0 +1,234 @@
+-- Quaries to write on meal table
+-- Get all meals
+SELECT
+  meal.title
+from
+  meal;
+-- Add a new meal
+INSERT INTO
+  meal(
+    `title`,
+    `description`,
+    `location`,
+    `when`,
+    `max_reservations`,
+    `price`,
+    `created_date`
+  )
+VALUES(
+    'dish of the day',
+    'the dish of the day is a choise of the main chef and is different efery day. Please ask for details',
+    'china town lyngby',
+    '2022-02-16 09:09:09',
+    200,
+    90,
+    '2022-02-15'
+  );
+-- Get a meal with any id, fx 1
+SELECT
+  -- meal.title
+  *
+FROM
+  meal
+WHERE
+  id = 1;
+-- Update a meal with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE
+  meal
+SET
+  meal.description = 'traditional spicy pizza'
+WHERE
+  meal.title LIKE '%pizza';
+-- Delete a meal with any id, fx 1
+DELETE FROM
+  meal
+WHERE
+  id = 1;
+-- Quries to write on Reservation table
+  -- Get all reservations
+select
+  meal.title as Food,
+  reservation.number_of_guests as 'Total Guests',
+  meal.when as 'Date of Event'
+from
+  reservation
+  JOIN meal on reservation.meal_id = meal.id;
+-- Add a new reservation
+INSERT INTO
+  reservation(
+    `number_of_guests`,
+    `created_date`,
+    `contact_phonenumber`,
+    `contact_name`,
+    `contact_email`,
+    `meal_id`
+  )
+VALUES(
+    85,
+    '2022-02-16',
+    '75757575',
+    'marta',
+    'marta@gmail.com',
+    4
+  );
+-- Get a reservation with any id, fx 1
+select
+  *
+from
+  reservation
+where
+  id = 2;
+SELECT
+  -- DISTINCT(meal.title)
+  *
+FROM
+  meal
+  JOIN reservation on meal.id = reservation.meal_id
+WHERE
+  meal_id = 2;
+-- Update a reservation with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE
+  reservation
+SET
+  contact_phonenumber = '27859433'
+WHERE
+  meal_id = 1;
+-- Delete a reservation with any id, fx 1
+DELETE FROM
+  reservation
+WHERE
+  meal_id = 1;
+-- Quries to write on Review table
+SELECT
+  title
+from
+  review;
+-- Add a new review
+INSERT INTO
+  review(
+    `title`,
+    `description`,
+    `stars`,
+    `created_date`,
+    `meal_id`
+  )
+VALUES(
+    'best food',
+    'i ate best food ever i found in my life. thanks to the main chef there. salute',
+    5,
+    '2022-02-26',
+    3
+  );
+-- Get a review with any id, fx 1
+SELECT
+  *
+from
+  review
+where
+  id = 1;
+-- Update a review with any id, fx 1. Update any attribute fx the title or multiple attributes
+UPDATE
+  review
+SET
+  stars = 5
+WHERE
+  meal_id = 2;
+-- Delete a review with any id, fx 1
+DELETE FROM
+  review
+WHERE
+  id = 1;
+-- Additional queries on functionnality
+  -- Get meals that has a price smaller than a specific price fx 90
+SELECT
+  meal.title,
+  meal.price
+from
+  meal
+WHERE
+  price = 90;
+-- Get meals that still has available reservations
+SELECT
```

Comment:
Remember that you need to sum the guests over all reservations and compare with max_reservations of the meal.

This is the trap query btw. There are at least two ways of doing it, both involve stuff that we didn't talk about in class:
 - use the COALESCE() function: https://dev.mysql.com/doc/refman/8.0/en/comparison-operators.html#function_coalesce
 - use GROUP BY, SUM, and `HAVING: https://www.w3schools.com/sql/sql_having.asp

Also remember that there might be meals that don't have any reservations yet.

Reply:
* help **help !! , could not figure it out 

SELECT
  meal.title,
  meal.max_reservations,
  reservation.number_of_guests
FROM
  meal
  JOIN reservation ON meal.id = reservation.meal_id
GROUP BY
  meal.title
HAVING
  SUM(reservation.number_of_guests) < meal.max_reservations;

Reply:
"how it should not take reservation more than max_reservations? is it JS logic ?" <- yes

the query looks good, except i don't see a reason to group by reservation.created_date
and (meal.max_reservations - sum(reservation.number_of_guests)) should be greater than 0, if it is 0 or less, it means you cannot make any more reservations.

does it make sense?